src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -119,21 +107,20 @@ private static IVsReferenceItems GetReferences(IImmutableDictionary<string, IImm
 
         private static IVsReferenceItems GetProjectReferences(IImmutableDictionary<string, IImmutableDictionary<string, string>> projectReferenceItems)
         {
-            IVsReferenceItems referenceItems = GetReferences(projectReferenceItems);
+            // NuGet expects each "<ProjectReference />" to have metadata item "ProjectFileFullPath" that represents
+            // the full path of the project reference, just use the built-in "FullPath" metadata for that.
+            IImmutableDictionary<string, IImmutableDictionary<string, string>> replacementProjectReferenceItems = projectReferenceItems;
```

Comment:
Using a builder here would likely involve fewer allocations. Might not matter if there aren't likely to be many changes.

Reply:
This code is deleted in a latter branch anyway.

Reply:
Going to do this in a followup.